librz/asm/arch/rl78/rl78.c

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: 2023 Bastian Engel <bastian.engel00@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include "rl78.h"
+
+#include "maps.h"
+
+static bool parse_operand(struct rl78_operand *operand, size_t *p,
+                          const ut8 *buf, size_t buf_len);
+
+static bool optype_es_applies(enum rl78_operand_type type);
+
+bool rl78_dis(struct rl78_instr RZ_OUT *instr, size_t RZ_OUT *bytes_read,
+              const ut8 *buf, size_t buf_len)
+{
+        if (buf_len == 0) {
+                *bytes_read = 0;
+                return false;
+        }
+
+        // points to the next byte to be processed
+        size_t p = 0;
+        int byte = buf[p++];
+
+        bool extension_addressing = false;
+        if (byte == 0x11) {
+                extension_addressing = true;
+                if (p >= buf_len) {
+                        *bytes_read = p;
+                        return false;
+                }
+
+                byte = buf[p++];
+        }
+
+        switch (byte) {
+                case 0x31:
+                        // 4th map
+                        *instr = rl78_instr_maps[3 * 256 + byte];
```

Comment:
Better do something like:
```
case X:
    map_base = 3 * 256;
case Y:
    map_base = 2 * 256;
...

}
*instr = rl78_instr_maps[map_base + byte];
```

It is easier to grasp what happens if you haven't looked at the table above.